cmd/syncthing/main.go

```diff
@@ -907,6 +901,32 @@ func autoUpgrade(cfg config.Wrapper, app *syncthing.App, evLogger events.Logger)
 	}
 }
 
+func initialAutoUpgrade(misc *db.NamespacedKV) (upgrade.Release, error) {
+	if last, ok, err := misc.Time(upgradeCheckKey); err != nil {
+		return upgrade.Release{}, err
```

Comment:
I think we should be more accepting of the database throwing an error, here and below. There being an unmarshal error on the time shouldn't be reason to block an upgrade, for example? We just avoid the upgrade if everything checks out and the times say we should avoid it, otherwise we try, imho.